### DISCUSSION TITLE
Help instrumenting with annotations

### DISCUSSION THREAD

**Discussion by oscaratienzaatosnet** [open-telemetry/opentelemetry-java-instrumentation#10903] [Q&A]
Hello! i am trying to instrument a class using my own annotations (the same as extended annotatioms) but neither extended annotations nor my own annotations work.

i think it is because some muzzle check is failing? 

failSafe(try(AnnotationInstrumentationModule#WithSpanInstrumentation((not(name(startsWith(kotlin.coroutines.))) and declaresMethods(whereOne((declaresAnnotations(whereOne(ofAnnotationType(name(equals(io.opentelemetry.instrumentation.annotations.WithSpan))))) and returns(erasure(is(class java.lang.Object))) and io.opentelemetry.javaagent.instrumentation.instrumentationannotations.KotlinCoroutineUtil$$Lambda$249/0x0000000800e46e88@408b35bf and not(false))))))) or false)

[otel.javaagent 2024-03-15 15:47:52:258 +0100] [main] DEBUG net.bytebuddy.agent.builder.AgentBuilderUtil - Could not decompose matcher failSafe(try(AnnotationInstrumentationModule#WithSpanInstrumentation(declaresMethods(whereOne(declaresAnnotations(whereOne(ofAnnotationType(name(equals(io.opentelemetry.instrumentation.annotations.WithSpan))))))))) or false)

[otel.javaagent 2024-03-15 15:47:52:258 +0100] [main] DEBUG net.bytebuddy.agent.builder.AgentBuilderUtil - Could not decompose matcher failSafe(try(AnnotationInstrumentationModule#AddingSpanAttributesInstrumentation(declaresMethods(whereOne((declaresAnnotations(whereOne(ofAnnotationType(name(equals(io.opentelemetry.instrumentation.annotations.AddingSpanAttributes))))) and not(declaresAnnotations(whereOne(ofAnnotationType(name(equals(io.opentelemetry.instrumentation.annotations.WithSpan))))))))))) or false)

[otel.javaagent 2024-03-15 15:47:52:258 +0100] [main] DEBUG net.bytebuddy.agent.builder.AgentBuilderUtil - Could not decompose matcher failSafe(try(ExternalAnnotationInstrumentationModule#ExternalAnnotationInstrumentation(declaresMethods(whereOne(declaresAnnotations(whereOne(ofAnnotationType(name(in(com.example.infrastructure.annotations.TraceCallAverage))))))))) or false)

[otel.javaagent 2024-03-15 15:47:52:259 +0100] [main] DEBUG net.bytebuddy.agent.builder.AgentBuilderUtil - Could not decompose matcher failSafe(try(WithSpanInstrumentationModule#WithSpanInstrumentation(declaresMethods(whereOne(declaresAnnotations(whereOne(ofAnnotationType(name(equals(io.opentelemetry.extension.annotations.WithSpan))))))))) or false)

[otel.javaagent 2024-03-15 15:47:52:259 +0100] [main] DEBUG net.bytebuddy.agent.builder.AgentBuilderUtil - Could not decompose matcher failSafe(try(RmiJpmsInstrumentationModule#ExposeRmiModuleInstrumentation((io.opentelemetry.javaagent.instrumentation.rmi.context.jpms.ExposeRmiModuleInstrumentation$1@737edcfa and name(startsWith(sun.rmi))))) or false)

[otel.javaagent 2024-03-15 15:47:52:262 +0100] [main] DEBUG net.bytebuddy.agent.builder.AgentBuilderUtil - Could not decompose matcher failSafe(try(SpringExtendedInstrumentationModule#SpringExtendedInstrumentation(declaresAnnotations(whereOne(ofAnnotationType(name(equals(com.example.infrastructure.annotations.TraceCallAverage))))))) or false

how can i debug this? i am trying to instrument just what i want to. without using WithSpan or so decorators.

**Comment by laurit** [comment]
Perhaps you could provide a minimal sample app that could be used to reproduce the issue you are having.

**Comment by oscaratienzaatosnet** [comment]
i am using a modified version of this (changing the decorator to my own) 

https://github.com/DGuhr/spring-otel-extended/tree/main/otel-javaagent-spring-extension

and i still dont have this working :( the settings i put in the instrumentation are this

      - name: OTEL_LOGS_EXPORTER
        value: otlp
      - name: OTEL_INSTRUMENTATION_COMMON_DEFAULT_ENABLED
        value: "true"
      - name: OTEL_INSTRUMENTATION_OPENTELEMETRY_API_ENABLED
        value: "true"
      - name: OTEL_INSTRUMENTATION_OPENTELEMETRY_INSTRUMENTATION_ANNOTATIONS_ENABLED
        value: "true"
      - name: OTEL_INSTRUMENTATION_EXAMPLE_EXTENDED_ENABLED
        value: "true"
      - name: OTEL_INSTRUMENTATION_EXTERNAL_ANNOTATIONS_ENABLED
        value: "true"
      - name: OTEL_INSTRUMENTATION_EXTERNAL_ANNOTATIONS_INCLUDE  #Dotel.instrumentation.external-annotations.include
        value: "com.example.infrastructure.annotations.TraceCallAverage"
      - name: OTEL_JAVAAGENT_DEBUG
        value: "true"
